physics/ephemeris.hpp

```diff
@@ -185,13 +185,11 @@ class Ephemeris {
 
   virtual Status last_severe_integration_status() const;
 
-  // Calls |ForgetBefore| on all trajectories.  On return |t_min() == t|.  This
-  // function is thread-hostile in the sense that it can cause |t_min()| to
-  // increase, so if it is called is parallel with code that iterates over the
-  // trajectories of the ephemeris, it can cause trouble.
-  // TODO(phl): Consider eliminating this function and truncating on
-  // serialization/deserialization.
-  virtual void ForgetBefore(Instant const& t) EXCLUDES(lock_);
+  // If the time |t| is not protected by a |Guard|, calls |ForgetBefore| on all
+  // trajectories and returns true, after which |t_min() == t|.  If the time |t|
+  // is protected by a |Guard|, returns false; the actual action is delayed
+  // until the destruction of the |Guard|.
+  virtual bool TryToForgetBefore(Instant const& t) EXCLUDES(lock_);
```

Comment:
This doesn't just try, it *will* eventually forget before `t`; `MarkForForgetting`?

In what circumstances would we use the result? `ForgetBefore` only exists for performance purposes, I can't think of a case where we would do something differently if we cannot forget immediately.

Reply:
1. I'd rather have an identifier that contains `ForgetBefore` for clarity.
1. In tests, it's convenient to check that the right thing happens.  Could also be useful in `CHECK`s.